src/BlackSlope.Api.Common/Extensions/BlackSlopeServiceCollectionExtensions.cs

```diff
@@ -1,62 +1,100 @@
+using System.IO;
+using System.Reflection;
+using AutoMapper;
+using BlackSlope.Api.Common.Configurtion;
+using BlackSlope.Api.Common.Swagger;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using Swashbuckle.AspNetCore.Swagger;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class BlackSlopeServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds Swagger service to the IServiceCollection and configure it 
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="swaggerConfig"></param>
+        /// <returns></returns>
+        public static IServiceCollection AddSwagger(this IServiceCollection services, SwaggerConfig swaggerConfig) => services.AddSwaggerGen(c =>
+            {
+
+                c.SwaggerDoc(swaggerConfig.Version, new Info { Title = swaggerConfig.ApplicationName, Version = swaggerConfig.Version });
+                c.AddSecurityDefinition("Bearer", new ApiKeyScheme()
+                {
+                    In = "header",
+                    Description = "Please insert JWT with Bearer into field",
+                    Name = "Authorization",
+                    Type = "apiKey"
+                });
+                c.DocumentFilter<DocumentFilterAddHealth>();
+                c.AddSecurityRequirement(new Dictionary<string, IEnumerable<string>>
+                {
+                    { "Bearer", new string[] { } }
+                });
+
+                // Set the comments path for the Swagger JSON and UI.
+                var xmlFile = swaggerConfig.XmlFile;
+                var xmlPath = Path.Combine(AppContext.BaseDirectory, xmlFile);
+                c.IncludeXmlComments(xmlPath);
+            });
+
+        /// <summary>
+        /// Adds MVC service to the Service Collection and configure json serializer behavior 
+        /// </summary>
+        /// <param name="services"></param>
+        /// <returns></returns>
+        public static IMvcBuilder AddMvcService(this IServiceCollection services) =>
+            services.AddMvc()
+                .SetCompatibilityVersion(CompatibilityVersion.Version_2_2)
+                .AddJsonOptions(options =>
+                {
+                    options.SerializerSettings.Converters.Add(new StringEnumConverter());
+                    options.SerializerSettings.NullValueHandling = NullValueHandling.Ignore;
+                });
+
+        /// <summary>
+        /// Add Azure service to the Service Collection and configure it
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="azureAdConfig"></param>
+        /// <returns></returns>
+        public static AuthenticationBuilder AddAzureAd(this IServiceCollection services, AzureAdConfig azureAdConfig) =>

```

Comment:
This is more of a meta level discussion, but do we want to host forked "child" branches that are tying us off to cloud specific services (ie - Azure, GCP, AWS, etc)?

Our "master" would be the golden agnostic copy, while any specific cloud target branches would inherit 1-branch-step down.

Reply:
It is a good point, and yes it is in our roadmap to push out these. This revision contains some improvements and more ready to do the segregation, but the actual process will be done as part of NuGetization. For example, we plan to add support for multiple identity providers and each could be integrated through a package.